redfish-core/lib/led.hpp

```diff
@@ -323,6 +323,23 @@ inline void
         std::variant<bool>(ledState));
 }
 
+inline bool validateLedState(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                             const std::string& ledState)
+{
+    if (ledState.empty())
+    {
+        return true;
+    }
+
+    if (ledState != "Lit" && ledState != "Blinking" && ledState != "Off")
```

Comment:
This is not how we normally do this..

Reply:
like this?
```
if (ledState == "Lit" || ledState == "Blinking" || ledState == "Off")
{
    return true;
}

return false;
```

Reply:
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/43761/15/redfish-core/lib/systems.hpp

Reply:
Something more along these lines 
```
inline void translatePowerMode(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
                               const std::string& modeValue)
{
    std::string modeString;

    if (modeValue == "xyz.openbmc_project.Control.Power.Mode."
                     "PowerMode.Static")
    {
        aResp->res.jsonValue["PowerMode"] = "Static";
    }
    else if (modeValue == "xyz.openbmc_project.Control.Power.Mode."
                          "PowerMode.MaximumPerformance")
    {
        aResp->res.jsonValue["PowerMode"] = "MaximumPerformance";
    }
    else if (modeValue == "xyz.openbmc_project.Control.Power.Mode."
                          "PowerMode.PowerSaving")
    {
        aResp->res.jsonValue["PowerMode"] = "PowerSaving";
    }
    else if (modeValue == "xyz.openbmc_project.Control.Power.Mode."
                          "PowerMode.OEM")
    {
        aResp->res.jsonValue["PowerMode"] = "OEM";
    }
    else
    {
        // Any other values would be invalid
        BMCWEB_LOG_DEBUG << "PowerMode value was not valid: " << modeValue;
        messages::internalError(aResp->res);
    }
}
```

Reply:
> Something more along these lines
> 
> ```
> inline void translatePowerMode(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
>                                const std::string& modeValue)
> {
>     std::string modeString;
> 
>     if (modeValue == "xyz.openbmc_project.Control.Power.Mode."
>                      "PowerMode.Static")
>     {
>         aResp->res.jsonValue["PowerMode"] = "Static";
>     }
>     else if (modeValue == "xyz.openbmc_project.Control.Power.Mode."
>                           "PowerMode.MaximumPerformance")
>     {
>         aResp->res.jsonValue["PowerMode"] = "MaximumPerformance";
>     }
>     else if (modeValue == "xyz.openbmc_project.Control.Power.Mode."
>                           "PowerMode.PowerSaving")
>     {
>         aResp->res.jsonValue["PowerMode"] = "PowerSaving";
>     }
>     else if (modeValue == "xyz.openbmc_project.Control.Power.Mode."
>                           "PowerMode.OEM")
>     {
>         aResp->res.jsonValue["PowerMode"] = "OEM";
>     }
>     else
>     {
>         // Any other values would be invalid
>         BMCWEB_LOG_DEBUG << "PowerMode value was not valid: " << modeValue;
>         messages::internalError(aResp->res);
>     }
> }
> ```

I don't think so, Maybe we are the same page:
The setLocationIndicatorActive method can set `LocationIndicatorActive` or `LedState`. If the last parameter ledState of this method is equal to empty, it means that this method is to set LocationIndicatorActive. If ledState is not equal to empty, it means to set LedState. At this time, we need to verify the LedState value. Whether the value is (Lit & Blinking & Off), if not, report an error directly.